features/prog.go

```diff
@@ -261,6 +270,8 @@ func haveProgramHelper(pt ebpf.ProgramType, helper asm.BuiltinFunc) error {
 		spec.AttachType = ebpf.AttachSkLookup
 	case ebpf.Syscall:
 		spec.Flags = sys.BPF_F_SLEEPABLE
+	case ebpf.Netfilter:
```

Comment:
Can you extend the unit tests to check at least one netfilter helper? We don't get coverage by default, I think.

Reply:
@lmb sorry, I'm a bit lost here. Looking at the helpers  I don't see any netfilter specific ones (yet). In the kernel selftests I see e.g. `bpf_dynptr_*` helpers [being used](https://web.git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/tree/tools/testing/selftests/bpf/progs/ip_check_defrag.c), but these helpers are used for other program types as well, so I'm not sure whether that's applicable.

Reply:
Correct, I don't think there are any netfilter-specific helpers. The last one (cgrp_storage_delete) was merged in Oct '22, which is before Netfilter progs were merged (April '23). Pretty normal given the decision to introduce new helper functionality using kfuncs instead; there likely won't be any netfilter-specific helpers, ever. And Kfuncs can be probed using CO-RE within the bpf prog itself, so no need for a Go API to probe them.

Reply:
@tklauser I think what Lorenz means is adding a test to `TestHaveProgramHelper` that exercises this `case ebpf.Netfilter`, since coverage here is not provided by the addition to `haveProgramTypeMatrix` above.

That said, it looks like we're missing tests for `CGroupSockopt`, `SkLookup` and `Syscall`.. Would you be able to pick those up in this PR as well? :pray:

Reply:
Done, thanks for the explanation. Pushed an additional commit adding coverage for `CGroupSockopt`, `SkLookup` and `Syscall`.